addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2822,6 +2822,220 @@ options.registry.ReplaceImage = options.Class.extend({
+        },
+        {
+            selectName: 'visibilityValueUtmMedium',
+            dataName: 'data-utm-medium',
+        },
+        {
+            selectName: 'visibilityValueUtmSource',
+            dataName: 'data-utm-source',
+        },
+        {
+            selectName: 'visibilityValueUtmCampaign',
+            dataName: 'data-utm-campaign',
+        },
+    ],
+    /**
+     * @constructor
+     */
+    init() {
+        this._super(...arguments);
+        if (this.$target[0].dataset.visibility === 'conditional') {
+            // hack to use the same visibility pannel as the popups
+            this.$target[0].classList.add('o_snippet_invisible');
+            this.trigger_up('snippet_option_visibility_update', { show: true });
+        }
+    },
+    /**
+     * Inserts or deletes record's id and value in target's data-attributes
+     * if no ids are selected, deletes the attribute
+     *
+     * @see this.selectClass for parameters
+     */
+    selectRecord(previewMode, widgetValue, params) {
+        const recordsData = JSON.parse(widgetValue);
+        const values = [];
+        for (const record of recordsData) {
+            values.push({ id: record.id, value: record[params.callWith] });
+        }
+        if (values.length) {
+            this.$target[0].dataset[params.attributeName] = JSON.stringify(values);
+        } else {
+            delete this.$target[0].dataset[params.attributeName];
+        }
+    },
+    /**
+     * Opens the toggler when 'conditionnal' is selected
+     *
+     * @override
+     */
+    selectDataAttribute(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === 'visibility' && widgetValue === 'conditional') {
+            const collapseEl = this.$el.children('we-collapse')[0];
+            this._toggleCollapseEl(collapseEl);
+            this.trigger_up('snippet_option_visibility_update', { show: true });
+        }
+    },
+    /**
+     * @override
+     */
+    async onTargetHide() {
+        this.$target[0].classList.add('d-none');
+    },
+    /**
+     * @override
+     */
+    async onTargetShow() {
+        this.$target[0].classList.remove('d-none');
+    },
+    /**
+     * Reads target's attributes and creates CSS selectors
+     * Stores them in data-attributes to then be reapplied by
+     * content/inject_dom.js
+     *
+     * @override
+    */
+    async cleanForSave() {
+        // removes classes that were used for the invisible elements pannel
+        this.$target[0].classList.remove('o_snippet_invisible');
+        this.$target[0].classList.remove('d-none');
+        if (this.$target[0].dataset.visibility !== 'conditional') {
+            return await this._super(...arguments);
+        }
+        /**
+         * There are 2 data attributes per option
+         * One that stores the current records selected
+         * Another that stores the value of the rule "Hide for / Visible for"
+         */
```

Comment:
We normally only use `/** */` style comments for method/class documentation, you can use a series of one line comments for those.